[ZT] Global WARP override (#20560)
cloudflare/cloudflare-docs: src/content/docs/cloudflare-one/connections/connect-devices/warp/configure-warp/warp-settings/index.mdx

2. Go to **My Team** > **Devices**.
3. Select **View** for a connected device.
4. Scroll down to **User details** and copy the 7-digit **Override code**.
5. Share this code with the user for them to enter on their device.

The user will have an unlimited amount of time to activate their code.

#### Enter the override code

To activate the override code on a user device:

1. In the WARP client, go to **Settings** > **Preferences** > **Advanced**.
2. Select **Enter code**.

> 3. Enter the override code. The WARP client will display a pop-up window showing when the override expires.

Reviewer: On windows, the override expiration time isn't displayed. CLIENT-11375 should hopefully address this.

> The user can now toggle the WARP switch or use the `warp-cli connect` command. The client will automatically reconnect after the [Auto connect period](#auto-connect), but the user can continue to turn on or off WARP until the override expires.

Reviewer: Yes, `warp-cli connect` will also be disabled if the GUI switch is locked (unless there's some override in place).

The CGNAT IP assigned to a WARP device is permanent until the device unregisters from your Zero Trust organization. Disconnects and reconnects do not change the IP address assignment.

### Global WARP override

<Details header="Feature availability">

| [WARP modes](/cloudflare-one/connections/connect-devices/warp/configure-warp/warp-modes/) | [Zero Trust plans](https://www.cloudflare.com/teams-pricing/) |
| ----------------------------------------------------------------------------------------- | ------------------------------------------------------------- |
| All modes                                                             | All plans                                                     |

| System   | Availability | Minimum WARP version |
| -------- | ------------ | -------------------- |

> | Windows  | ✅           | 2024.12.554.0        |

Author: ```suggestion
| Windows  | ✅           | 2025.2.600.0        |
```

> | macOS    | ✅           |                      |

Author: ```suggestion
| macOS    | ✅           |  2025.2.600.0                    |
```

> | Linux    | ✅           |                      |

Author: ```suggestion
| Linux    | ✅           |   2025.2.600.0                |
```

The CGNAT IP assigned to a WARP device is permanent until the device unregisters from your Zero Trust organization. Disconnects and reconnects do not change the IP address assignment.

### Global WARP override

<Details header="Feature availability">

| [WARP modes](/cloudflare-one/connections/connect-devices/warp/configure-warp/warp-modes/) | [Zero Trust plans](https://www.cloudflare.com/teams-pricing/) |
| ----------------------------------------------------------------------------------------- | ------------------------------------------------------------- |
| All modes                                                             | All plans                                                     |

| System   | Availability | Minimum WARP version |
| -------- | ------------ | -------------------- |
| Windows  | ✅           | 2024.12.554.0        |
| macOS    | ✅           |       |
| Linux    | ✅           |         |
| iOS      | ✅           | 1.8                  |
| Android  | ✅           |                      |
| ChromeOS | ✅           |                      |

</Details>

:::note
Requires the [Super Administrator](/cloudflare-one/roles-permissions/) role.
:::

> **Global WARP override** allows administrators to fail open WARP in case of an incident or outage. When you turn on **Global WARP override**, Cloudflare will immediately disconnect all WARP clients that are connected to your Zero Trust organization. End users will receive a notification on their device and the WARP client will display `The administrator for your account has disconnected WARP`.

Reviewer: There's a 4hrs SLA for all devices to pick the change (still in deliberation, but it's configurable). For most cases, this should happen instanteously. @kokolocomotion1 is it worth mentioning this?

Reviewer 2: The polling_frequency_seconds to configure the polling, was added to decouple the dependencies on the notification rearchitecture with the intention of retiring the field once account wide notifications were available. I'm not sure if we want to advertise the SLA as configurable :/

Author: Talked with Koko about this. We decided to hold off on mentioning the SLA, at least during the beta phase.

> All clients including end user devices, [WARP Connector](/cloudflare-one/connections/connect-networks/private-net/warp-connector/) hosts, and [WARP-to-WARP](/cloudflare-one/connections/connect-networks/private-net/warp-to-warp/) devices will be unable to connect until you turn off **Global WARP override**. [Auto connect](#auto-connect) settings and any active [Admin override](#admin-override) codes will not apply while the override is on.

Author: ```suggestion
All clients including end user devices, [WARP Connector](/cloudflare-one/connections/connect-networks/private-net/warp-connector/) hosts, and [WARP-to-WARP](/cloudflare-one/connections/connect-networks/private-net/warp-to-warp/) devices will be unable to connect until you turn off **Global WARP override**. [Auto connect](#auto-connect), [Lock WARP switch](#lock-warp-switch), and any active [Admin override](#admin-override) codes will not apply while the override is on.
```

> You can provide users with a new [Admin override](#admin-override) code to manually reconnect as needed during the override. For example, you may want IT staff to test resolution of the incident that led to the global disconnect.

Author: Does global override disable `warp-cli connect`?

> The only way for users to reconnect during a global override is by using a new [Admin override](#admin-override) code. For example, you can provide IT staff with a code to test resolution of the incident that led to the global disconnect.

Author: ```suggestion
The only way for users to reconnect during a global override is by using a new [Admin override](#admin-override) code. For example, you may want to provide IT staff with a code so that they can test resolution of the incident that led to the global disconnect.
```

> The only way for users to reconnect during a global override is by using a new [Admin override](#admin-override) code. For example, you may want to provide IT staff with a code so that they can test resolution of the incident that led to the global disconnect.

Reviewer: I think we should also call out that any existing admin overrides will be cleared when a global warp override is issued.

The CGNAT IP assigned to a WARP device is permanent until the device unregisters from your Zero Trust organization. Disconnects and reconnects do not change the IP address assignment.

### Global WARP override

<Details header="Feature availability">

| [WARP modes](/cloudflare-one/connections/connect-devices/warp/configure-warp/warp-modes/) | [Zero Trust plans](https://www.cloudflare.com/teams-pricing/) |
| ----------------------------------------------------------------------------------------- | ------------------------------------------------------------- |
| All modes                                                             | All plans                                                     |

| System   | Availability | Minimum WARP version |
| -------- | ------------ | -------------------- |

> | Windows  | ✅           | 2024.12.554.0        |

Author: need more info on the supported WARP versions

Reviewer: This isn't yet supported on all mobile platforms (there's follow up work for this).

Author: Plan is to add the GA version #.

| macOS    | ✅           |       |
| Linux    | ✅           |         |
| iOS      | ✅           | 1.8                  |
| Android  | ✅           |                      |
| ChromeOS | ✅           |                      |

</Details>

:::note
Requires the [Super Administrator](/cloudflare-one/roles-permissions/) role.
:::

> When you turn on **Global WARP override**, Cloudflare will immediately disconnect all WARP clients that are currently connected to your Zero Trust organization. This allows WARP to fail open in case of an incident or outage. [Auto connect](#auto-connect) will not apply while the override is on. Users will receive a notification on their device, and the WARP client will display `Error code: GLOBAL_WARP_OVERRIDE`.

Author: What happens if the device is not currently connected when you send the override signal?

If [Lock WARP switch](#lock-warp-switch) is `Disabled`, the user will be able to manually reconnect using the WARP GUI or the `warp-cli connect` command. This allows test users to come back online as needed to test resolution of the incident that led to the global disconnect. It also allows users who may not be impacted by the incident to resume normal operations.

> To immediately reconnect all WARP clients, turn off **Global WARP override**.

Author: Will this override signal reconnect clients that were manually disconnected?

> 

Author: Are there situations where we send the reconnect signal but the device is unable to reconnect?
